lrauv_ignition_plugins/scripts/make_dem_utm_tiles.py

```diff
@@ -0,0 +1,141 @@
+    :param tile_size: meters, each tile will have are of tile_size x tile_size
+    :param overlap: meters, tiles will have "overlap" meters shared with adjacent tiles on each edge
+    :return: void
+    """
+    unique_size = tile_size - 2 * overlap
+
+    # Open the file and create an instance of the ncCDF4 class
+    nc_fid = nc.Dataset(map_path, 'r')
+
+    # Extract data from NetCDF file
+    x = nc_fid.variables['x'][:]
+    y = nc_fid.variables['y'][:]
+    z = nc_fid.variables['z'][:]
+
+    # Find DEM boundary limits (to nearest 100)
+    x_lim = [np.ceil(min(x) / 100) * 100, np.floor(max(x) / 100) * 100]
+    y_lim = [np.ceil(min(y) / 100) * 100, np.floor(max(y) / 100) * 100]
+
+    # Arrange DEM tile corners
+    x_right_corner = np.arange(x_lim[1], x_lim[0] + tile_size, -unique_size - overlap).tolist()
+    y_bottom_corner = np.arange(y_lim[0], y_lim[1] - tile_size, unique_size + overlap).tolist()
+    # Combine to create the tile grid
+    [A, B] = np.meshgrid(x_right_corner, y_bottom_corner)
+    C = np.concatenate((A.T, B.T), axis=1)
+    bottom_right_corner = np.reshape(C, (int(C.size/2), 2), order='F')
+
+    # Tile boundaries:
+    left_bound  = bottom_right_corner[:, 0] - tile_size
+    right_bound = bottom_right_corner[:, 0]
+    lower_bound = bottom_right_corner[:, 1]
+    upper_bound = bottom_right_corner[:, 1] + tile_size
+
+    # Check the data
+    assert all(left_bound >= x_lim[0]), "Tiles violate min X bound."
+    assert all(right_bound <= x_lim[1]), "Tiles violate max X bound."
+    assert all(lower_bound >= y_lim[0]), "Tiles violate min Y bound."
+    assert all(upper_bound <= y_lim[1]), "Tiles violate max Y bound."
+
+    # Tile centers:
+    nTiles = len(bottom_right_corner)
+    x_tile_centers = (left_bound + right_bound) / 2
+    y_tile_centers = (lower_bound + upper_bound) / 2
+
+    P = Proj(proj='utm', zone=utm_zone, ellps='WGS84', preserve_units=True)
+    lon_tile_centers, lat_tile_centers = P(x_tile_centers, y_tile_centers, inverse=True)
+
+    # Log tile centers
+    tilesCenterFID = open(out_path + 'tile_centers.csv', 'w')
+    tilesCenterFID.write('Tile, UTM_zone, Eastings, Northings, Lat, Lon\n')
+
+    # Build gmt argument strings and loop through
+    gmt_grdcut = '/usr/local/bin/gmt grdcut ' + map_path + ' '
+
+    for i in range(nTiles):
+
+        # Assemble grdcut argument string
+        tile_name = 'PortugueseLedgeTile{}_UTM'.format(i)
+        tile_path = '{}{}.grd '.format(out_path, tile_name)
+        gcut_out_path = '-G{} '.format(tile_path)
+        gcut_bounds = '-R{}/{}/{}/{}'.format(left_bound[i], right_bound[i], lower_bound[i], upper_bound[i])
+
+        gmt_cmd = gmt_grdcut + gcut_out_path + gcut_bounds
+
+        # Write to centers file
+        tilesCenterFID.write(f'{tile_name}, {utm_zone}, '
+                             f'{x_tile_centers[i]}, {y_tile_centers[i]}, '
+                             f'{lat_tile_centers[i]}, {lon_tile_centers[i]}\n')
+
+        print(gmt_cmd)
```

Comment:
@braanan nit: leftover?

Reply:
Added some more text so it doesn't look random. These printouts help make sense of the GMT stdout